### PR TITLE
Speaker Mute/Unmute Automation Steps

### LINKED ISSUE #1250: Speaker Mute/Unmute Action
### Rules

- [X] I made myself familiar with the README.
- [X] I checked closed issues for similar ones and did not find a similar feature request mentioned before.
- [X] I understand that, if insufficient information was provided, my issue will be closed without an answer.

### Version

2.20.2

### OS

Windows 11, Version 23H2

### Device

Legion 515ACH6H

### Is your feature request related to a problem?

I have a problem where I want to add an action to connect and disconnect my charger so that I don't accidentally blast open a tab of YouTube I currently have open or any media for that matter. 

### How would you like the problem to be solved?

I want to add a step where you can mute and unmute your speaker when a trigger happened

### What alternatives have you considered?

I have tried some hacky ways with scripts but it didn't work

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by Ace-Radom**
Should close #1250 

Add automation step (`Speaker`) to automation step pools.

The function is implemented with `NAudio` lib, which has already been imported to the project.

Tested on:

- R9000P2021H (Windows11 23H2 Build 22631.3296)
- R9000P ARX8 (Windows10 22H2 Build 19045.4291)

## Summary by CodeRabbit

- **New Features**
	- Introduced speaker mute and unmute functionality, allowing users to automate these actions within the system.
	- Added user interface controls for managing speaker mute and unmute automation steps.

### CODE REVIEW

File: LenovoLegionToolkit.Lib.Automation/Steps/SpeakerMuteAutomationStep.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+public class SpeakerMuteAutomationStep : IAutomationStep
+{
+    public Task<bool> IsSupportedAsync() => Task.FromResult(true);
+
+    public Task RunAsync(AutomationContext context, AutomationEnvironment environment, CancellationToken token)
+    {
+        Speaker.Mute();
+        return Task.CompletedTask;
+    }
```
[the remark this answers is not in the document]
Comment (Ace-Radom) [reply LenovoLegionToolkit.Lib.Automation/Steps/SpeakerMuteAutomationStep.cs:15]: Should be good after the new commit

File: LenovoLegionToolkit.Lib.Automation/Steps/SpeakerUnmuteAutomationStep.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+public class SpeakerUnmuteAutomationStep : IAutomationStep
+{
+    public Task<bool> IsSupportedAsync() => Task.FromResult(true);
+
+    public Task RunAsync(AutomationContext context, AutomationEnvironment environment, CancellationToken token)
+    {
+        Speaker.Unmute();
+        return Task.CompletedTask;
+    }
```
[the remark this answers is not in the document]
Comment (Ace-Radom) [reply LenovoLegionToolkit.Lib.Automation/Steps/SpeakerUnmuteAutomationStep.cs:15]: Same above, should be good for now

**Comment by Ace-Radom** [comment]
@BartoszCichecki Done, ready for review.

Btw, I've noticed that these two steps always return a `CompletedTask`, which are the same as these in Turn on/off Wi-Fi steps. I don't know if the front-end can show sth went wrong when it happened, cuz I cannot produce an error-situation. Maybe it can be a problem.

**Comment by Ace-Radom** [comment]
@BartoszCichecki Done

**Review by BartoszCichecki — CHANGES_REQUESTED** [review]
Good work, however I think it would be enough to only keep the `SpeakerAllDevicesFeature` and remove the default variant. There is really no upside for having the "default device" variant and when connecting external devices in might cause confusion about which one is actually muted/unmuted. Additonally `SpeakerAllDevicesFeature` (and related classes) could be simply renamed to `SpeakerFeature` to align it with `MicrophoneFeature`.

**Review by BartoszCichecki — APPROVED** [review]

**Pull request merged**